internal/patch.js

```diff
@@ -7,19 +7,23 @@ module.exports = () => {
   Array.prototype[fl.equals] = function(y) {
     return this.length === y.length && this.join('') === y.join('');
   };
-  Array.prototype[fl.map] = Array.prototype.map;
+  Array.prototype[fl.map] = function(f) {
+    return this.map(x => f(x));
+  };
```

Comment:
In addition to mutating globally defined prototypes and featuring bugs such as the one above, this file contains a hack for `Array#fantasy-land/equals`, convincing me that the laws do not belong in this repository (#174). We really need [sanctuary-type-classes][1] to avoid monkey patching.

Sanctuary's property-based tests are really nice. See, for example, [__Functor.js__][2]. I'd like to move these tests to __fantasyland/fantasy-laws__ and remove the __laws__ directory from this repository. :)


[1]: https://github.com/sanctuary-js/sanctuary-type-classes
[2]: https://github.com/sanctuary-js/sanctuary/blob/master/test/internal/laws/Functor.js